subprojects/common/waiter/src/main/kotlin/com/avito/android/waiter/Waiter.kt

```diff
@@ -37,3 +37,19 @@ fun <T> waitFor(
 
     throw caughtAllowedException
 }
+
+fun repeatFor(
+    frequencyMs: Long = 50L,
+    timeoutMs: Long = TimeUnit.SECONDS.toMillis(2),
+    sleepAction: (frequencyMs: Long) -> Unit = { Thread.sleep(it) },
+    action: () -> Unit
+) {
+    var timer = 0L
+    val startTime = System.currentTimeMillis()
+
+    do {
+        action.invoke()
+        sleepAction(frequencyMs)
+        timer += frequencyMs
```

Comment:
If I'm not mistaken, "timer" can be deleted.
The check `System.currentTimeMillis() - startTime <= timeoutMs` already do the job.

Reply:
Yes, deleted it from the `waitFor` function too.